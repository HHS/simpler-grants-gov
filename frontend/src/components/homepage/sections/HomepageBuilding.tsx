import BuildingImage from "public/img/homepage-building.jpg";

import { useMessages, useTranslations } from "next-intl";
import Image from "next/image";
import { Grid, GridContainer } from "@trussworks/react-uswds";

const BuildingContent = () => {
  const t = useTranslations("Homepage.sections.building");
  const messages = useMessages() as unknown as IntlMessages;
  const { paragraphs } = messages.Homepage.sections.building;

  return (
    <GridContainer data-testid="homepage-building">
      <Grid row gap="md" className="padding-6">
        <Grid
          tablet={{
            col: true,
          }}
        >
          <Image
            alt="events-img"
            className="height-auto position-relative padding-right-6 padding-top-3"
            src={BuildingImage}
          />
        </Grid>
        <Grid
          tablet={{
            col: true,
          }}
        >
          <Grid>
            <h1>
              {t.rich("title", {
                span: (chunks) => <span className="text-italic">{chunks}</span>,
              })}
            </h1>
          </Grid>
          <Grid>
            {paragraphs.map((paragraph: string, paragraphIdx: number) => (
              <p
                className="line-height-sans-3 font-sans-md tablet:font-sans-lg text-balance"
                key={paragraphIdx}
              >
                {paragraph}
              </p>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </GridContainer>
  );
};

export default BuildingContent;
