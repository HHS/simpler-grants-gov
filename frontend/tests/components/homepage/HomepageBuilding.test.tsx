import { render, screen } from "tests/react-utils";

import HomepageBuilding from "src/components/homepage/sections/HomepageBuilding";

describe("Homepage Building Content", () => {
  it("matches snapshot", () => {
    const { container } = render(<HomepageBuilding />);
    expect(container).toMatchSnapshot();
  });
  it("Renders without errors", () => {
    render(<HomepageBuilding />);
    const component = screen.getByTestId("homepage-building");

    expect(component).toBeInTheDocument();
  });
});
