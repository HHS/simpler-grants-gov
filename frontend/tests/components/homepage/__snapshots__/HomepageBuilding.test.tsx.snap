// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Building Content matches snapshot 1`] = `
<div>
  <div
    class="grid-container padding-y-4 tablet-lg:padding-y-6"
    data-testid="homepage-building"
  >
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class="tablet-lg:grid-col tablet-lg:padding-right-6"
        data-testid="grid"
      >
        <img
          alt="events-img"
          class="height-auto position-relative margin-bottom-2 tablet-lg:margin-bottom-0"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <div
        class="tablet-lg:grid-col"
        data-testid="grid"
      >
        <div
          class=""
          data-testid="grid"
        >
          <h2>
            Building 
            <span
              class="text-italic"
            >
              with
            </span>
             you, not 
            <span
              class="text-italic"
            >
              for
            </span>
             you
          </h2>
          <p
            class="font-sans-md tablet-lg:font-sans-lg"
          >
            Transparency is the foundation of good government. That’s why we’re committed to sharing our process and working in the open.
          </p>
          <p
            class="font-sans-md tablet-lg:font-sans-lg"
          >
            All of our code is open-source and the roadmap is public. We welcome everyone to collaborate with us on the vision and details of every feature we build.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
